bubblegum/program/src/lib.rs

```diff
@@ -638,12 +644,16 @@ fn process_collection_verification<'info>(
     let collection_mint = ctx.accounts.collection_mint.to_account_info();
     let edition_account = ctx.accounts.edition_account.to_account_info();
     let collection_authority = ctx.accounts.collection_authority.to_account_info();
+    let collection_authority_record_pda = ctx
+        .accounts
+        .collection_authority_record_pda
+        .to_account_info();
     let bubblegum_signer = ctx.accounts.bubblegum_signer.to_account_info();
     let token_metadata_program = ctx.accounts.token_metadata_program.to_account_info();
 
     // Look for collection authority record PDA as a remaining account.
-    let collection_authority_record = if !ctx.remaining_accounts.is_empty() {
-        Some(&ctx.remaining_accounts[0])
+    let collection_authority_record = if collection_authority_record_pda.key() == crate::id() {
```

Comment:
The json docs for `collectionAuthorityRecordPda` mention "If there is no collecton authority record PDA then this must be the Bubblegum program address". Is setting `collection_authority_record` here to `None` when `collection_authority_record_pda.key()` is different from `crate::id()` in line with that? (seems a bit like the other way around at first, name it being `None` where that account is the program id)

Reply:
Yes you are correct I reversed the logic.